src/ape/managers/chain.py

```diff
@@ -450,6 +458,66 @@ def get(
 
         return contract_type or default
 
+    def resolve_proxy(self, address: AddressType) -> Optional[AddressType]:
+        code = self.provider.get_code(address).hex()[2:]
+        patterns = [
+            # eip-1167 minimal proxy contract
+            r"363d3d373d3d3d363d73(.{40})5af43d82803e903d91602b57fd5bf3",
+            # vyper <0.2.9 create_forwarder_to
+            r"366000600037611000600036600073(.{40})5af4602c57600080fd5b6110006000f3",
+            # 0xsplits clones
+            r"36603057343d52307f830d2d700a97af574b186c80d40429385d24241565b08a7c559ba283a964d9"
+            + r"b160203da23d3df35b3d3d3d3d363d3d37363d73(.{40})5af43d3d93803e605b57fd5bf3",
+        ]
+        for pattern in patterns:
+            match = re.match(pattern, code)
+            if match:
+                return self.conversion_manager.convert(match.group(1), AddressType)
+
+        slots = [
+            # eip-1967 standard proxy storage slots
+            int(self.provider.web3.keccak(text="eip1967.proxy.implementation").hex(), 16) - 1,
+            # openzeppelin upgradeability proxy
+            self.provider.web3.keccak(text="org.zeppelinos.proxy.implementation"),
+            # eip-1822 universal upgradeable proxy standard (uups)
+            self.provider.web3.keccak(text="PROXIABLE"),
+        ]
+        for slot in slots:
+            storage = self.provider.web3.eth.get_storage_at(address, slot)
+            if sum(storage) != 0:
+                return self.conversion_manager.convert(storage[-20:].hex(), AddressType)
+
+        # eip-1967 beacon proxy
+        slot = int(self.provider.web3.keccak(text="eip1967.proxy.beacon").hex(), 16) - 1
+        storage = self.provider.web3.eth.get_storage_at(address, slot)
+        if sum(storage) != 0:
+            abi = MethodABI(
+                type="function",
+                name="implementation",
+                stateMutability="view",
+                outputs=[ABIType(type="address")],
+            )
+            beacon = self.conversion_manager.convert(storage[-20:].hex(), AddressType)
+            return ContractCall(abi, beacon)()
+
+        # gnosis safe proxy
```

Comment:
Enumerating all the proxies in a doc would be nice! There are so many. Even if it happens in the doc str of the method, that is fine. Else, we may need `contracts.md` userguide (in `docs/userguides`). If you go that route, you don't have to thoroughly create it, just initialize it, add the enumeration, and we will fill it in later with non-relevant details to this PR.

Reply:
added `ProxyType` enum